app/src/components/RegisterArtifact.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  RegisterForm,
+  DEFAULT_ERRORS,
+  RegisterOnSubmit,
+  TextFields,
+  ErrorMessages,
+  FilesContext,
+  Document,
+} from './register/RegisterForm';
+import * as React from 'react';
+import Accordion from 'react-bootstrap/Accordion';
+import Card from 'react-bootstrap/Card';
+import FormSubmitButton from './register/FormSubmitButton';
+import { ArtistProvider } from '../providers/ArtistProvider';
+import { ContractProps } from '../helper/eth';
+import ArtistSelection from './register/ArtistSelection';
+import RegisterFields from './register/RegisterFields';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import DropZone from './register/DropZone';
+import ListGroup from 'react-bootstrap/ListGroup';
+import Button from 'react-bootstrap/Button';
+
+const registerArtifact: RegisterOnSubmit = async ({ fields }, _event): Promise<void> => {
+  console.log(fields);
+};
+
+const registerValidator: (textFields: TextFields) => ErrorMessages = (_fields) => {
+  return DEFAULT_ERRORS;
+};
+
+const FileList: React.FC = () => {
+  const { files } = React.useContext(FilesContext);
+  return <ListGroup>
+    {files.documents.map((document: Document, index: number) => {
+      return <ListGroup.Item key={index}>
+        <p>filename: {document.filename}</p>
+      </ListGroup.Item>;
+    })}
+  </ListGroup>;
+};
+
+const RegisterFieldLayout: React.FC = () => {
+  return <>
+    <Accordion defaultActiveKey="0">
+      <Card>
+        <Accordion.Toggle as={Card.Header} eventKey="0">
+          Artifact Information
+        </Accordion.Toggle>
+        <Accordion.Collapse eventKey="0">
+          <Card.Body>
+            <RegisterFields/>
+            <ArtistSelection/>
+          </Card.Body>
+        </Accordion.Collapse>
+      </Card>
+    </Accordion>
+    <FormSubmitButton/>
+  </>;
+};
+
+const ImageDropZone: React.FC = () => {
+  const { files, setFiles } = React.useContext(FilesContext);
+  return <DropZone popup callback={(hash: string) => {
+    setFiles({ ...files, image: hash });
```

Comment:
I'm pretty sure you said that this wasn't done yet: but commenting here that it should be done before merging